ACLWeb.js

```diff
@@ -34,175 +34,456 @@
 	***** END LICENSE BLOCK *****
 */
 
+var ext2mime = {
+	gz: 'application/gzip',
+	tgz: 'application/gzip',
+	pdf: 'application/pdf',
+	zip: 'application/zip',
+	tar: 'application/x-tar',
+	txt: 'text/plain',
+	rar: 'application/x-rar-compressed',
+	rtf:	'application/rtf',
+	bz2: 'application/x-bzip2',
+	bz: 'application/x-bzip',
+	doc: 'application/msword',
+	docx: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+	ppt: 'application/vnd.ms-powerpoint',
+	pptx: 'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+	html: 'text/html',
+	png: 'image/png',
+	gif: 'image/gif',
+	jpg: 'image/jpeg',
+	jpeg: 'image/jpeg',
+	mp4: 'video/mp4'
+};
+
 function detectWeb(doc, url) {
-	if (doc.contentType === 'application/pdf' || url.endsWith('.bib')) {
-		let id = url.split('/').pop();
-		return id[0] == 'J' || id[0] == 'Q' ? 'journalArticle' : 'conferencePaper';
-	} else {
+	if (doc.contentType === 'application/pdf' || url.endsWith('.bib') || url.includes('/papers/')) {
+		if (url.endsWith('/')) {
+			url = url.slice(0, -1);
+		}
+		let id = url.split('/').pop().toLowerCase();
+		return id[0] == 'j' || id[0] == 'q' ? 'journalArticle' : 'conferencePaper';
+	} else if ((url.includes('/events/') || url.includes('/people/')
+		|| url.includes('/volumes/') || url.includes('/search/'))
+		&& getSearchResults(doc, url)) {
 		return 'multiple';
 	}
+	return false;
 }
 
 function doWeb(doc, url) {
 	if (detectWeb(doc, url) === 'multiple') {
-		Zotero.selectItems(extractFullProceedings(doc), function (selected) {
+		Zotero.selectItems(getSearchResults(doc, url), function (selected) {
 			if (!selected) {
```

Comment:
@zuphilip -- I'm understanding correctly that this simply isn't needed (anymore) and can be removed together with the `return true; inserted further down to fix linting?

Reply:
As far as I know, the `selectItems` callback doesn't expect a return value, so `return true` shouldn't be needed.

And I would hope that the argument to the callback is always an object, such that `if (!selected)` isn't necessary. If that's not the case we should fix that.

Reply:
If nothing is checked in the selection dialog, the callback in `selectItems` is called with `null`, so `Object.keys` gives a `TypeError: can't convert null to object`. 

I have cleaned up that code wrt. returns. The build on this new commit failed because of linting problems with other translators, not sure why it ran those.

Reply:
Oh, that's a shame. We should fix that and make a note of what version has the fix so we can do a cut-off down the line and stop including the extra lines. I've created an [issue](https://github.com/zotero/zotero/issues/1705). Thanks.